File: backend/app/api/owners/createOwner/CreateOwnerController.java
package api.owners.createOwner;

import com.fasterxml.jackson.databind.JsonNode;
import play.libs.Json;
import play.libs.concurrent.HttpExecutionContext;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Results;

import javax.inject.Inject;
import java.util.concurrent.CompletionStage;

public class CreateOwnerController extends Controller {

    @Inject
    CreateOwnerService ownerService;

    @Inject
    HttpExecutionContext ec;


    public CompletionStage<Result> createOwner() throws Exception {
        JsonNode json = request().body().asJson();
        System.out.println("json = " + json);
        OwnerCreateJava c = Json.mapper().readValue(json.toString(), OwnerCreateJava.class);
        System.out.println("c = " + c);

        final OwnerCreateJava resource = Json.fromJson(json, OwnerCreateJava.class);

        return ownerService.create(resource.firstName(), resource.lastName(), resource.address(), resource.city(), resource.telephone())
                .handle((savedResource, error) -> {
                    if (error != null) {
                        return Results.badRequest();
                    } else {
                        return Results.created(Json.toJson(savedResource));
                    }
                });
    }

}

File: backend/modules/petclinic/app/petclinic/api/owners/createOwner/CreateOwnerController.scala
package petclinic.api.owners.createOwner

import javax.inject.Inject

import play.api.libs.json.Json
import play.api.mvc.{AbstractController, ControllerComponents, Results}

import scala.compat.java8.FutureConverters._

class CreateOwnerController @Inject()(service: CreateOwnerService, cc: ControllerComponents) extends AbstractController(cc) {
  implicit val ec = cc.executionContext


  def createOwner = cc.actionBuilder.async(cc.parsers.json) { request =>

    val owner = request.body.as[OwnerCreate]

    service
      .create(owner.firstName, owner.lastName, owner.address, owner.city, owner.telephone)
      .toScala
      .map(o => Results.Created(Json.toJson(o)))
      .recover {
        case x => Results.BadRequest
      }
  }

}
